### PR TITLE
test: mysql datatypes cypress test cases

### LINKED ISSUE #16331: [Task]: Datatype cypress test cases for mySql plugin
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

Write cypress test cases for all datatypes to be handled by mySql plugin
| Index  | Datatype | True Cases | False Cases |
| ------------- | ------------- | ------------- | ------------- |
| 1  | Boolean  | Completed | Completed |
| 2  | Enum  | Completed | Completed |
| 3  | Signed-TinyInt  | Completed | Completed |
| 4  | Unsigned-TinyInt  | Completed | Completed |
| 5  | Signed-SmallInt  | Completed | Completed |
| 6  | Unsigned-SmallInt  | Completed | Completed |
| 7  | Signed-MediumInt  | Completed | Completed |
| 8  | Unsigned-MediumInt  | Completed | Completed |
| 9  | Signed-Int  | Completed | Completed |
| 10  | Unsigned-Int  | Completed | Completed |
| 11 | BigInt  | Completed | Completed |
| 12 | Float  | Completed | Completed |
| 13 | Double  | Completed | Completed |
| 14 | Decimal | Completed | Completed |
| 15 | DateTime  | Completed | Completed |
| 16  | JSON  | Completed | Completed |
| 17 | Char/Varchar  | Completed | Completed |

### PR DESCRIPTION
**Pull request by ChandanBalajiBP**
## Description

> Cypress test cases for mysql datatypes.

Fixes #16331 

## Type of change
- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?
- Cypress

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/client/cypress/integration/Smoke_TestSuite/ServerSideTests/Datatypes/MySQL_Spec.ts [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+        table.ReadTableRowColumnData(j, i, 0).then(($cellData) => {
+          expect($cellData).to.eq(value);
+        });
+      });
+    });
+  });
+
+  it("8. Deleting all records from table ", () => {
+    agHelper.GetNClick(locator._deleteIcon);
+    agHelper.AssertElementVisible(locator._spanButton("Run InsertQuery"));
```
Comment (Aishwarya-U-R) [inline app/client/cypress/integration/Smoke_TestSuite/ServerSideTests/Datatypes/MySQL_Spec.ts:129]: For Deleting also - is the button 'Run InsertQuery'?

File: app/client/cypress/integration/Smoke_TestSuite/ServerSideTests/Datatypes/MySQL_Spec.ts [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    cy.wait(2000);
+    inputData.result.forEach((res_array, i) => {
+      res_array.forEach((value, j) => {
+        table.ReadTableRowColumnData(j, i, 0).then(($cellData) => {
+          expect($cellData).to.eq(value);
+        });
+      });
+    });
+  });
+
```
[the remark this answers is not in the document]
Comment (ChandanBalajiBP) [reply app/client/cypress/integration/Smoke_TestSuite/ServerSideTests/Datatypes/MySQL_Spec.ts:116]: Have DSL ready for the update also. But : 
- For testing datatypes, the query type wouldn't affect much.
- Already the test takes about 10min on average, so didn't want to push more load. 
- Will think of a better approach for this and update the same while addressing the next plugin.

File: app/client/cypress/integration/Smoke_TestSuite/ServerSideTests/Datatypes/MySQL_Spec.ts [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+  //Insert false values to each column and check for the error status of the request.
+  it("5. False Cases", () => {
+    ee.ActionTemplateMenuByEntityName(inputData.databaseName, "INSERT");
+    agHelper.RenameWithInPane("falseCases");
+    inputData.falseResult.forEach((res_array, i) => {
+      res_array.forEach((value) => {
+        query = `INSERT INTO datatypes (${inputData.inputFieldName[i]}) VALUES (${value})`;
+        dataSources.EnterQuery(query);
+        dataSources.RunQuery(false);
```
[the remark this answers is not in the document]
Comment (ChandanBalajiBP) [reply app/client/cypress/integration/Smoke_TestSuite/ServerSideTests/Datatypes/MySQL_Spec.ts:81]: 1. Yes
2. Here the type of error is not much of use, only the query failing would suffice. Also took this approach to reduce the runtime.

File: app/client/cypress/integration/Smoke_TestSuite/ServerSideTests/Datatypes/MySQL_Spec.ts [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+      res_array.forEach((value, j) => {
+        table.ReadTableRowColumnData(j, i, 0).then(($cellData) => {
+          expect($cellData).to.eq(value);
+        });
+      });
+    });
+  });
+
+  it("8. Deleting all records from table ", () => {
+    agHelper.GetNClick(locator._deleteIcon);
```
Comment (Aishwarya-U-R) [inline app/client/cypress/integration/Smoke_TestSuite/ServerSideTests/Datatypes/MySQL_Spec.ts:118]: Is this _deleteIcon mapped to deleteAllRecords? ThenI think you can remove 'deleteRecord' query creation & deletion steps aswell.

Comment (ChandanBalajiBP) [reply app/client/cypress/integration/Smoke_TestSuite/ServerSideTests/Datatypes/MySQL_Spec.ts:118]: Yes, I will remove those steps. Thank you.

**Review by Aishwarya-U-R — COMMENTED** [review]
Cypress looks good, Update query scripts will be handled with better approach as mentioned.,

**Review by Aishwarya-U-R — APPROVED** [review]
Since only cypress script changes exists, Approving!

**Pull request merged**